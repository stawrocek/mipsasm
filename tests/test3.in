#this test checks all supported instructions
#I compared results to MARS output

.text
main:
lui		$1 10
addi		$t4, $zero, 12
addiu		$t5, $sp, 10
slti		$s0, $zero, -10000
sltiu		$s1, $s0, 1337
andi		$s4, $gp, 123
ori 		$s4, $s6, 101
xori		$s7, $v0, 0xAAAA

sll		$s2, $s2, 4
srl		$s3, $s2, 2
sra		$s1, $t8, 3
sllv		$s0, $t9, $k0
srlv		$s0, $s1, $k1
srav		$s0, $s1, $s3
mfhi		$s0
mthi		$s1
mflo		$s2
mtlo		$s3
mult		$s4, $s5
multu 		$s6, $s7
div		$s7, $t0
divu		$t1, $t2
add		$t3, $t4, $t5
addu		$t6, $t7, $t8
and		$t9, $a0, $a1
or 		$a2, $a3, $k0
xor 		$s0, $gp, $sp
nor		$s1, $k1, $fp
slt		$s2, $k0, $ra
sltu		$s2, $k0, $ra

