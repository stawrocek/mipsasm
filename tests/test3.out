.text
main:
00400000	3c01000a		lui		$1 10
00400004	200c000c		addi		$t4, $zero, 12
00400008	27ad000a		addiu		$t5, $sp, 10
0040000c	2810d8f0		slti		$s0, $zero, -10000
00400010	2e110539		sltiu		$s1, $s0, 1337
00400014	3394007b		andi		$s4, $gp, 123
00400018	36d40065		ori 		$s4, $s6, 101
0040001c	3857aaaa		xori		$s7, $v0, 0xAAAA
00400020	00129100		sll		$s2, $s2, 4
00400024	00129882		srl		$s3, $s2, 2
00400028	001888c3		sra		$s1, $t8, 3
0040002c	03598004		sllv		$s0, $t9, $k0
00400030	03718006		srlv		$s0, $s1, $k1
00400034	02718007		srav		$s0, $s1, $s3
00400038	00008010		mfhi		$s0
0040003c	02200011		mthi		$s1
00400040	00009012		mflo		$s2
00400044	02600013		mtlo		$s3
00400048	02950018		mult		$s4, $s5
0040004c	02d70019		multu 		$s6, $s7
00400050	02e8001a		div		$s7, $t0
00400054	012a001b		divu		$t1, $t2
00400058	018d5820		add		$t3, $t4, $t5
0040005c	01f87021		addu		$t6, $t7, $t8
00400060	0085c824		and		$t9, $a0, $a1
00400064	00fa3025		or 		$a2, $a3, $k0
00400068	039d8026		xor 		$s0, $gp, $sp
0040006c	037e8827		nor		$s1, $k1, $fp
00400070	035f902a		slt		$s2, $k0, $ra
00400074	035f902b		sltu		$s2, $k0, $ra


